sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/src/QuestionAnsweringClient.cs

```diff
@@ -0,0 +1,179 @@
+        /// </summary>
+        /// <param name="endpoint">The Question Answering endpoint on which to operate.</param>
+        /// <param name="credential">A <see cref="AzureKeyCredential"/> used to authenticate requests to the <paramref name="endpoint"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
+        public QuestionAnsweringClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint, credential, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QuestionAnsweringClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">The Question Answering endpoint on which to operate.</param>
+        /// <param name="credential">A <see cref="AzureKeyCredential"/> used to authenticate requests to the <paramref name="endpoint"/>.</param>
+        /// <param name="options">Optional <see cref="QuestionAnsweringClientOptions"/> to customize requests sent to the endpoint.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
+        public QuestionAnsweringClient(Uri endpoint, AzureKeyCredential credential, QuestionAnsweringClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            Endpoint = endpoint;
+            options ??= new QuestionAnsweringClientOptions();
+
+            Diagnostics = new ClientDiagnostics(options);
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                new AzureKeyCredentialPolicy(credential, AuthorizationHeader));
+
+            _knowledgebaseRestClient = new(Diagnostics, Pipeline, Endpoint, options.Version);
+            _textRestClient = new(Diagnostics, Pipeline, Endpoint, options.Version);
+        }
+
+        /// <summary>
+        /// Protected constructor to allow mocking.
+        /// </summary>
+        protected QuestionAnsweringClient()
+        {
+        }
+
+        /// <summary>
+        /// Get the service endpoint for this client.
+        /// </summary>
+        public virtual Uri Endpoint { get; }
+
+        /// <summary>
+        /// Gets the <see cref="ClientDiagnostics"/> for this client.
+        /// </summary>
+        private protected virtual ClientDiagnostics Diagnostics { get; }
+
+        /// <summary>
+        /// Gets the <see cref="HttpPipeline"/> for this client.
+        /// </summary>
+        private protected virtual HttpPipeline Pipeline { get; }
+
+        /// <summary>Answers the specified question using your knowledgebase.</summary>
+        /// <param name="projectName">The name of the project to use.</param>
+        /// <param name="options">The question to answer.</param>
```

Comment:
my understanding is that there are other properties that can be set in here right? Consider saying something along those lines too